spec/rubocop/cop/lint/mixed_regexp_capture_types_spec.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Lint::MixedRegexpCaptureTypes do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  it 'registers an offense when both of named and numbered captures are used' do
+    expect_offense(<<~RUBY)
+      /(?<foo>bar)(baz)/
+      ^^^^^^^^^^^^^^^^^^ Do not mix named captures and numbered captures in a Regexp literal.
+    RUBY
+  end
+
+  it 'does not register offense to a regexp with named capture only' do
+    expect_no_offenses(<<~RUBY)
+      /(?<foo>bar)/
```

Comment:
I think it's better to use multiple named captures instead of a single named capture.

```suggestion
      /(?<foo>foo?<bar>bar)/
```